src/actions/secrets.js

```diff
@@ -0,0 +1,81 @@
+/*
+Copyright 2019 The Tekton Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import { createCredential, deleteCredential, getCredentials } from '../api';
+import { getSelectedNamespace } from '../reducers';
+
+export function fetchSecretsSuccess(data) {
+  return {
+    type: 'SECRETS_FETCH_SUCCESS',
+    data
+  };
+}
+
+export function fetchSecrets({ namespace } = {}) {
+  return async (dispatch, getState) => {
+    dispatch({ type: 'SECRETS_FETCH_REQUEST' });
+    let secrets;
+    try {
+      const selectedNamespace = namespace || getSelectedNamespace(getState());
+      secrets = await getCredentials(selectedNamespace);
+      const secretsFormatted = [];
+      secrets.forEach((secret, index) => {
+        const object = {
+          uid: index.toString(),
```

Comment:
remove this as it is likely to cause issues if used

Reply:
currently the uid is being used as a key in the table which means ill have to make another PR on that. can I add those changes in this PR or separate one?

Reply:
`uid` shouldn't be used anywhere and should be removed from this PR.

Since the table's not being used yet, the change to the table can be made separately. If you need a unique value as a `key` then `namespace:name` would work and you should already have that info available where you need it.